src/PluginDataModel.h

```diff
@@ -0,0 +1,80 @@
+/*!*********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  @file PluginDataModel.h
+
+  @author Vitaly Sverchinsky
+
+**********************************************************************/
+#pragma once
+
+#include <wx/dataview.h>
+
+#include "PluginDescriptor.h"
+
+class PluginDescriptor;
+
+///!brief A plugins list model that can be attached to wxDataViewCtrl
+class PluginDataModel final : public wxDataViewModel
+{
+   static const void* RowIdRoot;
+
+   std::vector<std::pair<PluginDescriptor*, bool>> mPluginStateModel;
+   std::vector<void*> mIndexFilterMap;
+
+   wxString mFilterType;
+   wxString mFilterExpr;
+   int mFilterState{-1};
+   int mFilterCategory{-1};
+
+public:
+
+   enum Column
+   {
+      ColumnName,
+      ColumnType,
+      ColumnPath,
+      ColumnState,
+      ColumnNum
+   };
+
+   PluginDataModel(int filterEffectCategory = -1,
+                   int filterState = -1,
+                   const wxString& filterType = {},
+                   const wxString& filterExpr = {});
+
+   wxDataViewItem GetItemForRow(uint32_t row) const;
```

Comment:
Either this should be an `override` or it is not used.
Please review this for other methods as well.
That will help the review, and making as many of these `private` as possible on top of that would help even more.